### PR TITLE
Check dummy inputs order and correct it with wrapper

### PR DESCRIPTION
**Pull request by JingyaHuang**
*Related issue: #18 and https://github.com/aws-neuron/aws-neuron-sdk/issues/642*

__Context__

`torch_neuronx.trace` accepts tuples as inputs for the tracing, whereas we need to ensure that the dummy inputs tuple corresponds to the order of `model.forward()`. 

This PR will fix the previous unusual behavior of Flaubert and XLM models (where `token_type_ids` was mistaken as `langs` during the compilation).

### CODE REVIEW

File: optimum/exporters/neuron/base.py [hunk 1]
```diff
@@ -237,3 +238,31 @@ def generate_dummy_inputs(self, return_tuple=True, **kwargs) -> Union[Dict[str,
             return tuple(dummy_inputs.values())
         else:
             return dummy_inputs
+
+    def check_model_inputs_order(
+        self,
+        model: PreTrainedModel,
+        dummy_inputs: Dict[str, torch.Tensor],
```
Comment (michaelbenayoun) [inline optimum/exporters/neuron/base.py:245]:
```suggestion
        dummy_inputs: Dict[str, "torch.Tensor"],
```

Comment (JingyaHuang) [reply optimum/exporters/neuron/base.py:245]: `torch` is imported so imo there is no need to use quotes here.

File: optimum/exporters/neuron/base.py [hunk 2]
```diff
@@ -209,13 +210,13 @@ def outputs(self) -> List[str]:
...
 
-    def generate_dummy_inputs(self, return_tuple=True, **kwargs) -> Union[Dict[str, "torch.Tensor"], Tuple]:
+    def generate_dummy_inputs(self, return_tuple=False, **kwargs) -> Union[Dict[str, "torch.Tensor"], Tuple]:
         """
         Generates dummy inputs that the exported model should be able to process.
         This method is actually used to determine the input specs and their static shapes that are needed for the export.
 
         Returns:
-            `Dict[str, tf.Tensor]`: A dictionary mapping input names to dummy tensors.
+            `Dict[str, torch.Tensor]`: A dictionary mapping input names to dummy tensors.
```
[the remark this answers is not in the document]
Comment (JingyaHuang) [reply optimum/exporters/neuron/base.py:219]: Will add!

File: optimum/exporters/neuron/base.py [hunk 3]
```diff
@@ -209,13 +210,13 @@ def outputs(self) -> List[str]:
         """
         return self._TASK_TO_COMMON_OUTPUTS[self.task]
 
-    def generate_dummy_inputs(self, return_tuple=True, **kwargs) -> Union[Dict[str, "torch.Tensor"], Tuple]:
+    def generate_dummy_inputs(self, return_tuple=False, **kwargs) -> Union[Dict[str, "torch.Tensor"], Tuple]:
```
[the remark this answers is not in the document]
Comment (JingyaHuang) [reply optimum/exporters/neuron/base.py:213]: Will add typing for `return_tuple`!

File: optimum/exporters/neuron/base.py (hunk 1 above)
Comment (michaelbenayoun) [reply optimum/exporters/neuron/base.py:244]: Then you should remove present somewhere else in the code!

Comment (JingyaHuang) [reply optimum/exporters/neuron/base.py:244]: Yeah, quotes for `torch.Tensor` were removed here [36f2dbd](https://github.com/huggingface/optimum-neuron/pull/33/commits/36f2dbd804db355459c8665b2cd4733e44d41295). My bad, I was not careful enough on typing and docstrings in the exporter PR.

File: optimum/exporters/neuron/base.py [hunk 4]
```diff
@@ -209,13 +209,13 @@ def outputs(self) -> List[str]:
         """
         return self._TASK_TO_COMMON_OUTPUTS[self.task]
 
-    def generate_dummy_inputs(self, return_tuple=True, **kwargs) -> Union[Dict[str, "torch.Tensor"], Tuple]:
+    def generate_dummy_inputs(self, return_tuple: bool = False, **kwargs) -> Union[Dict[str, torch.Tensor], Tuple]:
```
Comment (michaelbenayoun) [inline optimum/exporters/neuron/base.py:212]: Add typing for the tuple

File: optimum/exporters/neuron/base.py [hunk 5]
```diff
@@ -209,13 +209,13 @@ def outputs(self) -> List[str]:
...
         """
         Generates dummy inputs that the exported model should be able to process.
         This method is actually used to determine the input specs and their static shapes that are needed for the export.
 
         Returns:
-            `Dict[str, tf.Tensor]`: A dictionary mapping input names to dummy tensors.
+            `Union[Dict[str, torch.Tensor], Tuple]`: A dictionary mapping input names to dummy tensors.
```
Comment (michaelbenayoun) [inline optimum/exporters/neuron/base.py:218]: Same here
```suggestion
            `Union[Dict[str, torch.Tensor], Tuple]`: A dictionary or tuple ? mapping input names to dummy tensors.
```

**Review by michaelbenayoun — APPROVED** [review]
Left a few nits, LGTM otherwise!

**Pull request merged**